docs/theming/developers.md

```diff
@@ -21,42 +22,185 @@ order: 4
+components community to support the passing of data between components.
+
+Our system utilizes this protocol with the setting of the `color-palette`
+attribute on a provider element which passes it's context data of `light` or
+`dark` on to it's children.  By doing so we can ensure accessible colors are
+applied given any possible change in context value higher up in the DOM tree. 
+
+This is important not only for helping us ensure we maintain our own design
+guidelines, but also for accessibility compliance and enabling great experiences
+for all our users.
+
+The context protocol is enabled by two [reactive controllers][controllers]: the
+[provider controller][providersrc], and the [consumer controller][consumersrc].
+These controllers work in tandem to alternately provide the context data, or
+consume it as a child element. Custom elements can implement the consumer, the
+provider, or both, depending on the needs of that particular element.
+
+#### Providers
+
+Custom Elements that implement the provider controller are elements that provide
+their own context, overriding that of their parent.  Elements such as
+`<rh-surface>`, `<rh-card>`, and `<rh-accordion>` are examples of such context
+providers. If a provider contains a set `color-palette` attribute, it will
+override any parent context, and pass it's context on to its children.
+
+To make your element a color context provider,
+
+1. First import the provider controller.
+  <rh-code-block dedent language="js" highlighting="client">
+    <script type="sample/javascript">
+      import { colorContextProvider, type ColorPalette } from '../../lib/context/color/provider.js';
+    </script>
+  </rh-code-block>
+2. Then add the `@colorContextProvider` decorator to a property with the
+   attribute `color-palette` which is the type `ColorPalette`.  
+  <rh-code-block dedent language="js" highlighting="client">
+    <script type="sample/javascript">
+      @colorContextProvider()
+      @property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette;
+    </script>
+  </rh-code-block>
+
+Read "[What are color palettes][palettes]" for more information about the sixe
+available `color-palettes`.
+
+#### Consumers
+
+Context consumers are elements that passively consume the parent context.
+Elements such as `<rh-cta>`, `<rh-badge>` and `<rh-tag>` are examples of
+consumers.
+
+To make your element a color context consumer,
+
+<ol>
+  <li>First `import` the classMap Lit directive, and the consumer controller
+    <rh-code-block dedent language="js" highlighting="client">
+      <script type="sample/javascript">
+        import { classMap } from 'lit/directives/class-map.js';
+        import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
+      </script>
+    </rh-code-block>
+  </li>
+  <li>Then add the `@colorContextConsumer()` decorator to a private property of `on` which is the type of `ColorTheme`.
```

Comment:
Change the three backticks on this line to use actual `<code>` HTML elements for proper rendering.